### PR TITLE
Hardwire return values for domain limits of `arcsin` and `arccos`

### PR DESCRIPTION
**Pull request by CedarGroveStudios**
fix for issue #7

### CODE REVIEW

File: jepler_udecimal/utrig.py [hunk 1]
```diff
@@ -198,7 +198,13 @@ def asin(x, context=None):
 
     with localcontext(context) as ctx:
         ctx.prec += 2
-        r = atan(x / (1 - x * x).sqrt())
+        pi = Decimal(1).atan() * 4
+        if x == 1:
+            r = pi / 2  # pi * 1/2 radians
+        elif x == -1:
+            r = pi * 3 / 2  # pi * 3/2 radians
```
Comment (jepler) [inline jepler_udecimal/utrig.py:205]: Shouldn't it be -pi/2?

File: jepler_udecimal/utrig.py [hunk 2]
```diff
@@ -218,10 +224,15 @@ def acos(x, context=None):
 
     with localcontext(context) as ctx:
         ctx.prec += 2
-        r = atan((1 - x * x).sqrt() / x)
+        if x == 1:
+            r = Decimal(0). # 0 radians
```
Comment (jepler) [inline jepler_udecimal/utrig.py:228]: Looks like a stray dot here? Or am I missing something?

File: jepler_udecimal/utrig.py [hunk 3]
```diff
@@ -218,10 +224,15 @@ def acos(x, context=None):
...
+        elif x == -1:
+            r = Decimal(1).atan() * 4  # pi radians
+        else:
+            r = atan((1 - x * x).sqrt() / x)
         if r < 0:
             r += 4 * atan(1)
-        return r
+        return r / 1
```
Comment (jepler) [inline jepler_udecimal/utrig.py:235]: It's a preexisting bug but shouldn't the return r/1 be outside the local context with statement?

**Review by jepler — CHANGES_REQUESTED** [review]
Thank you! I have some comments and possibly changes to request

File: jepler_udecimal/utrig.py (hunk 3 above)
Comment (CedarGroveStudios) [reply jepler_udecimal/utrig.py]: Yes. Fixed in the latest. Thank you for your keen eye.

File: jepler_udecimal/utrig.py (hunk 2 above)
Comment (CedarGroveStudios) [reply jepler_udecimal/utrig.py]: Yes, stray dot. Operator error. The GitHub editor inserted a period at the end of a sentence after two spaces are typed.

File: jepler_udecimal/utrig.py (hunk 1 above)
Comment (CedarGroveStudios) [reply jepler_udecimal/utrig.py:205]: Using the unit circle method, 3/2*pi rad is equal to -1/2*pi rad

Comment (jepler) [reply jepler_udecimal/utrig.py:205]: I'd still rather see it match math.asin:
```python
>>> math.asin(-1)
-1.5707963267948966
```

**Comment by jepler** [comment]
I believe your change to the return-precision of acos was correct, and the test cases need to be updated to reflect the change:
```
Failed example:
    Decimal('.1').acos()
Expected:
    Decimal('1.47062890563333682288579851219')
Got:
    Decimal('1.470628905633336822885798512')
```

File: jepler_udecimal/utrig.py (hunk 1 above)
Comment (CedarGroveStudios) [reply jepler_udecimal/utrig.py]: Good idea. Hadn't considered compatibility with `math.asin`. Thanks for your patience.

**Comment by CedarGroveStudios** [comment]
updated the `acos` example with the new value.

**Pull request merged**